pkg/kubelet/status/status_manager.go

```diff
@@ -898,12 +900,12 @@ func (m *manager) canBeDeleted(pod *v1.Pod, status v1.PodStatus, podIsFinished b
 		return false
 	}
 	// Delay deletion of pods until the phase is terminal.
-	if !podutil.IsPodPhaseTerminal(pod.Status.Phase) {
+	if !podutil.IsPodPhaseTerminal(status.Phase) {
 		klog.V(3).InfoS("Delaying pod deletion as the phase is non-terminal", "phase", status.Phase, "pod", klog.KObj(pod), "podUID", pod.UID)
 		return false
 	}
 	// If this is an update completing pod termination then we know the pod termination is finished.
-	if podIsFinished {
+	if podIsFinished || m.podDeletionSafety.PodIsFinished(pod) {
```

Comment:
Is there a place in status_manager_test to add a scenario driven unit test for this?  Would more clearly describe the issue.

Reply:
not currently, but I have added a preliminary `TestSyncPodForDeletedPodThatIsNeverStartedByPodWorkers`. It mocks up the `podDeletionSafetyProvider.PodIsFinished` (but `NeverStarted` would be enough in this scenario).

Reply:
Marking as resolved as this is outdated code change, as the current approach (of using HandlePodCleanups) does not modify this.